src/main/scala/cognite/spark/v1/FilesRelation.scala

```
@@ -91,7 +91,7 @@ class FilesRelation(config: RelationConfig)(val sqlContext: SQLContext)
 
   override def delete(rows: Seq[Row]): IO[Unit] = {
     val deletes = rows.map(r => fromRow[DeleteItem](r))
-    val ids = deletes.map(_.id)
+    val ids = deletes.flatMap(_.id)
```

In these cases, we should either have a separate DeleteItem without externalId field or at least make sure that it's not set. Otherwise, people will think that it works, but then it will silently ignore the delete requests.

---

I keep the old class `DeleteItem`, we use a new class `DeleteItemByCogniteId` for this ;)